Realm/RLMGeospatial.h

```diff
@@ -0,0 +1,253 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import <Realm/RLMConstants.h>
+
+RLM_HEADER_AUDIT_BEGIN(nullability)
+/// Conforming protocol for a Geo-shape.
+@protocol RLMGeospatial
+@end
+
+/**
+ A class that represents the coordinates of a point formed by a latitude and a longitude value.
+
+  * Latitude ranges between -90 and 90 degrees, inclusive.
+  * Longitude ranges between -180 and 180 degrees, inclusive.
+  * Altitude cannot have negative values.
+
+ Values outside this ranges will return nil when trying to create a `RLMGeospatialPoint`.
+
+ @note There is no dedicated type to store Geospatial points, instead points should be stored as
+ [GeoJson-shaped](https://www.mongodb.com/docs/manual/reference/geojson/) 
+ embedded object, as explained below. Geospatial queries (`geoWithin`) can only be executed
+ in such a type of objects and will throw otherwise.
+
+ Persisting geo points in Realm is currently done using duck-typing, which means that any model class with a specific **shape**
+ can be queried as though it contained a geographical location. The recommended approach is using an embedded object.
+
+ @warning This structure cannot be persisted and can only be used to build other geospatial shapes
+ such as (`RLMGeospatialBox`, `RLMGeospatialPolygon` and `RLMGeospatialCircle`).
+
+ @warning Altitude is not used in any of the query calculations.
+ */
+RLM_SWIFT_SENDABLE
+@interface RLMGeospatialPoint : NSObject
+/// Latitude in degrees.
+@property (readonly) double latitude;
+/// Longitude in degrees.
+@property (readonly) double longitude;
+/// Altitude distance.
+@property (readonly) double altitude;
```

Comment:
`altitude` should be nullable

Reply:
double cannot be nullable, in case of no value we set 0